docs/topics/api-guidelines-backward-compatibility.md

```diff
@@ -0,0 +1,318 @@
+* It is desirable but very difficult to guarantee source compatibility. As a library author, you must consider every 
+possible way a function or type could be invoked or instantiated by a user of the library.
+Source compatibility is typically an aspiration, not a promise.
+
+The rest of this section describes actions you can take, and tools you can use to help ensure the different kinds of compatibility.
+
+## Compatibility types {initial-collapse-state="collapsed"}
+
+**Binary compatibility** means that a new version of a library can replace a previously compiled version of the library.
+Any software that was compiled against the previous version of the library should continue to work correctly.
+
+> Learn more about binary compatibility in the [Binary compatibility validator’s README](https://github.com/Kotlin/binary-compatibility-validator?tab=readme-ov-file#what-makes-an-incompatible-change-to-the-public-binary-api) or from the [Evolving Java-based APIs](https://github.com/eclipse-platform/eclipse.platform/blob/master/docs/Evolving-Java-based-APIs-2.md) document.
+>
+{type="tip"}
+
+**Source compatibility** means that a new version of a library can replace a previous one without modifying any of the
+source code that uses the library. However, the outputs from compiling this client code may no longer
+be compatible with the outputs from compiling the library, so the client code must be rebuilt against the new version of
+the library to guarantee compatibility.
+
+**Behavioral compatibility** means that a new version of the library does not modify the existing functionality,
+except to fix bugs. The same features are involved and they have the same semantics.
+
+## Use the Binary compatibility validator
+
+JetBrains provides a [Binary compatibility validator](https://github.com/Kotlin/binary-compatibility-validator) tool, which can be used to ensure binary compatibility across different versions of your API.
+
+This tool is implemented as a Gradle plugin, and it adds two tasks to your build:
+
+* The `apiDump` task creates a human-readable `.api` file that describes your API.
+* The `apiCheck` task compares the saved description of the API to classes compiled in the current build.
+
+The `apiCheck` task is invoked at build time by the standard Gradle `check` task.
+When compatibility is broken, the build fails. At that point, you should run the `apiDump` task manually and compare
+the differences between the old and new versions.
+If you are satisfied with the changes, you can update the existing `.api` file, which resides within your VCS.
+
+The validator has [experimental support for validating KLibs](https://github.com/Kotlin/binary-compatibility-validator?tab=readme-ov-file#experimental-klib-abi-validation-support) produced by multiplatform libraries.
+
+## Specify return types explicitly
+
+As discussed in the [Kotlin coding guidelines](https://kotlinlang.org/docs/coding-conventions.html#coding-conventions-for-libraries),
```

Comment:
I think that relative links should work here (coding-conventions.md)

Reply:
Look at other integrated docs, I guess you'll find the answer

Reply:
good point - I checked it seems to be working - I replaced all occurences